ui/models/filter.js

```diff
@@ -0,0 +1,263 @@
+    this.toggleFilter('classifiedState', classifiedState);
+  }
+
+  toggleUnclassifiedFailures() {
+    if (this._isUnclassifiedFailures()) {
+      this.resetNonFieldFilters();
+    } else {
+      this.urlParams.resultStatus = [...thFailureResults];
+      this.urlParams.classifiedState = ['unclassified'];
+      this.push();
+    }
+  }
+
+  replaceFilter(field, value) {
+    this.urlParams[field] = !Array.isArray(value) ? [value] : value;
+    this.push();
+  }
+
+  clearNonStatusFilters() {
+    const { repo, resultStatus, classifiedState } = this.urlParams;
+
+    this.urlParams = { repo, resultStatus, classifiedState };
+    this.push();
+  }
+
+  /**
+   * reset the non-field (checkbox in the ui) filters to the default state
+   * so the user sees everything.  Doesn't affect the field filters.  This
+   * is used to undo the call to ``setOnlyUnclassifiedFailures``.
+   */
+  resetNonFieldFilters() {
+    const { resultStatus, classifiedState } = thFilterDefaults;
+
+    this.urlParams.resultStatus = [...resultStatus];
+    this.urlParams.classifiedState = [...classifiedState];
+    this.push();
+  }
+
+  /**
+   * Whether or not this job should be shown based on the current filters.
+   *
+   * @param job - the job we are checking against the filters
+   */
+  showJob(job) {
+    // when runnable jobs have been added to a resultset, they should be
+    // shown regardless of settings for classified or result state
+    const status = getStatus(job);
+    if (status !== 'runnable') {
+      // test against resultStatus and classifiedState
+      if (!this.urlParams.resultStatus.includes(status)) {
+        return false;
+      }
+      if (!this._checkClassifiedStateFilters(job)) {
+        return false;
+      }
+    }
+    // runnable or not, we still want to apply the field filters like
+    // for symbol, platform, search str, etc...
+    return this._checkFieldFilters(job);
+  }
+
+  _checkClassifiedStateFilters(job) {
+    const { classifiedState } = this.urlParams;
+    const isJobClassified = isClassified(job);
+
+    if (!classifiedState.includes('unclassified') && !isJobClassified) {
+      return false;
+    }
+    // If the filters say not to include classified, but it IS
+    // classified, then return false, otherwise, true.
+    return !(!classifiedState.includes('classified') && isJobClassified);
+  }
+
+  _checkFieldFilters(job) {
+    return Object.entries(this.urlParams).every(([field, values]) => {
+      let jobFieldValue = this._getJobFieldValue(job, field);
+
+      // If ``job`` does not have this field, then don't filter.
+      // Consider it a pass.  i.e.: runnable jobs have no ``tier`` field.
+      if (jobFieldValue) {
+        // All filter values are stored as lower case strings
+        jobFieldValue = String(jobFieldValue).toLowerCase();
+
+        switch (thFieldChoices[field].matchType) {
+
+          case thMatchType.substr:
+            if (!values.some(val => jobFieldValue.includes(val))) {
```

Comment:
Is this supposed to equal `false`? Maybe I'm missing something but it looks like its flipping the boolean that `some` returns.

Reply:
Yeah, dealing with double-negatives always gets me tangled.  

In this case, what this is doing is : If there isn't at least one of our specified values for this field that is a substring of this job's value, then return false.

Reply:
Makes sense looking at it now with a fresh mind rather than end of day :) thanks for humoring me.